arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -1415,51 +1377,57 @@ Result IResearchDataStore::initDataStore(
   };
 
   auto openFlags = irs::OM_APPEND;
-  if (!_dataStore._reader) {
+  if (!pathExists) {
     openFlags |= irs::OM_CREATE;
   }
 
-  _dataStore._writer = irs::index_writer::make(*(_dataStore._directory),
-                                               _format, openFlags, options);
+  _dataStore._writer = irs::IndexWriter::Make(*(_dataStore._directory), _format,
+                                              openFlags, options);
 
   if (!_dataStore._writer) {
     return {TRI_ERROR_INTERNAL,
-            absl::StrCat("failed to instantiate data store writer with path '",
+            absl::StrCat("Failed to instantiate data store writer with path '",
                          _dataStore._path.string(),
                          "' while initializing ArangoSearch index '",
                          index().id().id(), "'")};
   }
 
-  if (!_dataStore._reader) {
-    _dataStore._writer->commit();  // initialize 'store'
-    _dataStore._reader = irs::directory_reader::open(*(_dataStore._directory),
-                                                     _format, readerOptions);
+  if (!pathExists) {
+    _dataStore._writer->Commit();  // Initialize 'store'
   }
 
+  _dataStore._reader = _dataStore._writer->GetSnapshot();
+
   if (!_dataStore._reader) {
     _dataStore._writer.reset();
 
     return {TRI_ERROR_INTERNAL,
-            absl::StrCat("failed to instantiate data store reader with path '",
+            absl::StrCat("Failed to instantiate data store reader with path '",
                          _dataStore._path.string(),
                          "' while initializing ArangoSearch index '",
                          index().id().id(), "'")};
   }
 
-  if (!readTick(_dataStore._reader.meta().meta.payload(),
+  if (!readTick(irs::GetPayload(_dataStore._reader.Meta().index_meta),
                 _dataStore._recoveryTickLow, _dataStore._recoveryTickHigh)) {
     return {TRI_ERROR_INTERNAL,
-            absl::StrCat("failed to get last committed tick while initializing "
+            absl::StrCat("Failed to get last committed tick while initializing "
                          "ArangoSearch index '",
                          index().id().id(), "'")};
   }
 
-  LOG_TOPIC("7e128", TRACE, TOPIC)
-      << "data store reader for link '" << index().id()
-      << "' is initialized with recovery tick low '"
+  LOG_TOPIC("7e028", TRACE, TOPIC)
+      << "Successfully opened existing data store data store reader for "
```

Comment:
Is the log message now still appropriate? It says "opened existing data store", but I think we get here even if no data store existed before? Should the message depend on `pathExists`, e.g.
```suggestion
if (pathExists) {
  LOG_TOPIC(...) << "Successfully opened existing data store ..."
} else {
  LOG_TOPIC(...) << "Successfully initialized new data store ..."
}
```

Reply:
Done